frontend/src/routes/posts/ViewUserFeed.tsx

```diff
@@ -1,13 +1,114 @@
+    <><b>{count}</b> {`${word}${(count !== 1) ? 's' : ''}`}</>;
+
+type LoadState = 'INIT' | 'LOADED' | 'ERROR' | 'NOUSER';
 
 const ViewUserFeed: React.FC = () => {
-    const {username} = useParams();
-    return (
-        <div>
-            User: {username}
-        </div>
-    );
+    const { username } = useParams();
+    const { state: context } = React.useContext(Context);
+    const navigate = useNavigate();
+    const [loadState, setLoadState] = React.useState<LoadState>('INIT');
+    const [userInfo, setUserInfo] = React.useState<User>();
+    const [posts, setPosts] = React.useState<PostBasicInfo[]>([]);
+
+    const handleFollow = () => {
+        if (userInfo && context.loggedIn && username !== context.currentUser) {
+            if (userInfo.amFollowing)
+                setUserInfo({ ...userInfo, amFollowing: false });
+            else setUserInfo({ ...userInfo, amFollowing: true });
+        }
+        // TODO API call...
+    };
+
+    // Edit account button
+    const handleEditAccount = () => {
+        navigate('/accounts/edit');
+    };
+
+    // Infinite scrolling callback
+    const handleInfiniteScroll = () => {
+        setPosts(posts.concat(generateMockPosts(10)));
+    };
+
+    React.useEffect(() => {
+        setLoadState('INIT');
+
+        // TODO API call...
+        if (username?.includes('error'))
+            setLoadState('ERROR');
+        else if (username?.includes('noacc'))
+            setLoadState('NOUSER');
+        else {
+            setUserInfo({
+                ...mockUserInfo,
+                username: username || ''
+            });
+            setPosts(generateMockPosts(10));
+        }
+    }, [username]);
+
+    return <>
+        <section className="userfeed-profile-info">
+            <InfiniteScroll callback={handleInfiniteScroll} />
+            {userInfo && <ProfilePhoto user={userInfo} isMyProfile={context.currentUser === username} />}
+            <div className="userfeed-info">
+                {loadState === 'ERROR' && <p>Sorry, something went wrong...</p>}
+                {loadState === 'NOUSER' && <p>The requested user does not exist.</p>}
+                {userInfo && <>
+                    <h1>{username} - <em>{userInfo.realName}</em></h1>
+                    <ul className='userfeed-info-list'>
+                        <li>{pluralHelper('follower', userInfo.numberOfFollowers)}</li>
+                        <li><b>{userInfo.numberFollowing}</b> following</li>
+                        <li>{pluralHelper('post', userInfo.numberOfPosts)}</li>
+                    </ul></>}
+            </div>
+            <div className='userfeed-buttons'>
+                {context.loggedIn
+                    && context.currentUser !== username
+                    && <Button text={(userInfo?.amFollowing) ? 'Unfollow' : 'Follow'} onClick={handleFollow} />}
+                {context.currentUser === username
+                    && <Button text='Edit account' onClick={handleEditAccount} />}
```

Comment:
`<Button>` ima više smisla unutar forme koja treba da se šalje. Pošto ovo nije forma, već se koristi isključivo u svrhe navigacije, predložio bih sledeće:
```jsx
<Link to="/accounts/edit" className="button">Edit account</Link>
```
s tim što će `.button` možda morati da se modifikuje da radi i na linkovima. Posle ovoga ni `handleEditAccount` ni `navigate`/`useNavigate` neće biti potrebni.